### PR TITLE
Document support for PKCS#12

### PR DESCRIPTION
**Pull request by JakeSCahill**
## Description

Review deadline: 20 November

Note to reviewers: I haven't updated the Kubernetes docs because the Helm chart requires updates to support PKCS#12. Related Jira is here: https://redpandadata.atlassian.net/browse/K8S-347

## Page previews

https://deploy-preview-860--redpanda-docs-preview.netlify.app/24.3/manage/security/encryption/#pkcs

## Checks

- [x] New feature
- [ ] Content gap
- [ ] Support Follow-up
- [ ] Small fix (typos, links, copyedits, etc)

### CODE REVIEW

File: modules/manage/pages/security/encryption.adoc [hunk 1]
```diff
@@ -10,39 +10,48 @@ By default, Redpanda data is sent unencrypted. A security best practice is to en
...
+TLS certificates are required for encryption. You can use your own certificates, either self-signed or issued by a trusted CA.
+
+You'll need:
 
-* key file (`broker.key`)
-* certificate file (`broker.crt`)
-* truststore file (`ca.crt`). 
+- A private key file (`broker.key`) for each broker.
+- A certificate file (`broker.crt`) for each broker.
+- A Truststore file (`ca.crt`). All brokers can have the same `ca.crt` file.
```
Comment (JakeSCahill) [inline modules/manage/pages/security/encryption.adoc]:
```suggestion
- A truststore file (`ca.crt`). All brokers can have the same `ca.crt` file.
```

**Review by michael-redpanda — CHANGES_REQUESTED** [review]
Looks great!  Now the fun part - turns out PKCS#12 doesn't work in FIPS.  So I think we need a call out for that.

So some background information on this (discovered in [this PR](https://github.com/redpanda-data/redpanda/pull/24094))

The MAC key derivation algorithm used for PKCS#12 is PKCS12KDF.  This is _not_ a FIPS supported algorithm.  This means that, when running in FIPS mode, the OpenSSL FIPS provider will error out on unwrapping a PKCS#12 file.

A proposal ([rfc9579](https://datatracker.ietf.org/doc/html/rfc9579)) has been made to replace that with PBMAC1 which is FIPS compliant.  The ability to unwrap a PKCS#12 file using that is in OpenSSL 3.4, which we haven't updated to yet.

Some other reading can be found here: https://www.redhat.com/en/blog/fips-140-3-changes-pkcs-12

**Comment by JakeSCahill** [comment]
> Looks great! Now the fun part - turns out PKCS#12 doesn't work in FIPS. So I think we need a call out for that.
> 
> So some background information on this (discovered in [this PR](https://github.com/redpanda-data/redpanda/pull/24094))
> 
> The MAC key derivation algorithm used for PKCS#12 is PKCS12KDF. This is _not_ a FIPS supported algorithm. This means that, when running in FIPS mode, the OpenSSL FIPS provider will error out on unwrapping a PKCS#12 file.
> 
> A proposal ([rfc9579](https://datatracker.ietf.org/doc/html/rfc9579)) has been made to replace that with PBMAC1 which is FIPS compliant. The ability to unwrap a PKCS#12 file using that is in OpenSSL 3.4, which we haven't updated to yet.
> 
> Some other reading can be found here: https://www.redhat.com/en/blog/fips-140-3-changes-pkcs-12

Thanks @michael-redpanda - I added some of this context to the TLS doc and the FIPS doc

File: modules/manage/pages/security/encryption.adoc [hunk 2]
```diff
@@ -314,11 +334,12 @@ Schema Registry and HTTP Proxy connect to Redpanda over the Kafka API. If you co
...
 * xref:./listener-configuration.adoc[Configure Listeners]
 
+[[pkcs]]
 === Configure TLS with PKCS#12 bundles
 
-You can simplify certificate management by using a PKCS#12 bundle.
+You can simplify certificate management by generating a password-protected PKCS#12 bundle from your `broker.key` and `broker.crt` files.
 
-NOTE: If you haven't generated the `.key` and `.crt` files, refer to <<gen-certs, Generate certificate files>>.
+NOTE: PKCS#12 keys are not supported when xref:manage:security/fips-compliance.adoc[FIPS mode] is enabled in Redpanda. The PKCS12KDF algorithm used in PKCS#12 is not FIPS-compliant. To use Redpanda in FIPS mode, configure your certificates and keys in <<pem,PEM format>> instead.
```
Comment (michael-redpanda) [inline modules/manage/pages/security/encryption.adoc:342]: praise: 👍

**Review by michael-redpanda — APPROVED** [review]

File: modules/get-started/pages/whats-new.adoc [hunk 3]
```diff
@@ -49,7 +49,17 @@ This release includes several updates to xref:get-started:licensing/overview.ado
 
 - *30-day trial Enterprise license*: Starting with version 24.3, new Redpanda clusters automatically receive a trial license that's valid for 30 days, allowing unrestricted use of Enterprise features. This evaluation period begins when the cluster is created for the first time. After this period expires, inactive Enterprise features are disabled, and active features enter a restricted state.
 
-== Declarative user and ACL management in Kubernetes
+== PKCS#12 support for TLS listeners
+
+Redpanda now supports configuring TLS using PKCS#12 files, allowing you to manage both the certificate and private key in a single, password-protected file.
```
Comment (asimms41) [inline modules/get-started/pages/whats-new.adoc]:
```suggestion
Redpanda now supports configuring TLS using PKCS#12 files, which allows you to manage both the certificate and private key in a single, password-protected file.
```

File: modules/manage/pages/security/encryption.adoc [hunk 4]
```diff
@@ -55,47 +73,42 @@ default_md      = sha256
...
-# Common policy for nodes and users.
 [ signing_policy ]
 organizationName = supplied
 commonName       = optional
 ----
 
-Create a CA key to self-sign certificates:
-
-[,bash]
+. Create the CA's private key and set its permissions:
```
Comment (asimms41) [inline modules/manage/pages/security/encryption.adoc]: The rendering of the apostrophe on the preview looks a little odd.

File: modules/manage/pages/security/encryption.adoc [hunk 5]
```diff
@@ -113,31 +126,29 @@ certificates (created above).
 
 |`-batch` |Batch mode, where certificates are certified automatically.
 |===
++
+The output `ca.crt` is the CA's public certificate, which you'll use in the truststore.
```
Comment (asimms41) [inline modules/manage/pages/security/encryption.adoc:130]: Again, the rendering of the apostrophe looks odd?

File: modules/manage/pages/security/encryption.adoc [hunk 6]
```diff
@@ -221,38 +215,52 @@ applying extensions.
...
+
 == Configure TLS
 
-To configure TLS, in `redpanda.yaml`, enter:
+To configure TLS, in `redpanda.yaml`, enter either the standard <<pem,PEM configuration files>> or the <<pkcs,PKCS#12 bundle configuration>>.
+
+Choose PEM files when:
+
+- You are using FIPS mode compliance.
+- You prefer file-based configurations with separate key, certificate, and truststore file.
```
Comment (asimms41) [inline modules/manage/pages/security/encryption.adoc]:
```suggestion
- You prefer file-based configurations with a separate key, certificate, and truststore file.
```

File: modules/manage/pages/security/encryption.adoc [hunk 7]
```diff
@@ -221,38 +215,52 @@ applying extensions.
...
+
+Choose PKCS#12 bundles when:
+
+- FIPS mode is not required in your environment.
+- You want a single, password-protected file that contains all certificates and keys.
+
+[[pem]]
+=== Configure TLS with PEM files
+
+If you have separate `key_file`, `cert_file`, and `truststore_file`, use the following configuration in `redpanda.yaml`:
```
Comment (asimms41) [inline modules/manage/pages/security/encryption.adoc]:
```suggestion
If you have separate files for `key_file`, `cert_file`, and `truststore_file`, use the following configuration in `redpanda.yaml`:
```

File: modules/manage/pages/security/encryption.adoc [hunk 8]
```diff
@@ -375,9 +414,9 @@ redpanda:
   - name: internal
     enabled: true
     require_client_auth: true
-    cert_file: <path-to-PEM-formatted-cert-file>
-    key_file: <path-to-PEM-formatted-key-file>
-    truststore_file: <path-to-PEM-formatted-CA-file>
+    cert_file: <path-to-pem-cert-file>
+    key_file: <path-to-pem-key-file>
+    truststore_file: <path-to-pem-CA-file>
```
Comment (asimms41) [inline modules/manage/pages/security/encryption.adoc]: The caps in CA are stopping this from being an editable field.

**Review by asimms41 — APPROVED** [review]
Just a couple of minor comments/observations.

File: modules/manage/pages/security/encryption.adoc (hunk 4 above)
Comment (JakeSCahill) [inline modules/manage/pages/security/encryption.adoc]:
```suggestion
. Create a private key for the CA and set its permissions:
```

File: modules/manage/pages/security/encryption.adoc (hunk 5 above)
Comment (JakeSCahill) [inline modules/manage/pages/security/encryption.adoc:130]:
```suggestion
The output `ca.crt` is the CA's public certificate, which you'll use in the truststore.
```

File: modules/manage/pages/security/encryption.adoc (hunk 8 above)
Comment (JakeSCahill) [inline modules/manage/pages/security/encryption.adoc]:
```suggestion
    cert_file: <path-to-pem-cert-file>
    key_file: <path-to-pem-key-file>
    truststore_file: <path-to-pem-ca-file>
```

**Pull request merged**